Documentation/bootstrapping-guidelines.md

```diff
@@ -0,0 +1,111 @@
+
+To build .NET SDK from source -  
+ build.sh (args - path to the source built SDK???)
+
+## Building for New OS (Supported, RID Existing)
+
+**Supported OS**
+[Check](https://learn.microsoft.com/en-us/dotnet/core/install/linux) officially supported linux distributions.  
+[List](https://github.com/dotnet/installer/blob/release/7.0.1xx/src/SourceBuild/Arcade/eng/common/templates/job/source-build-run-tarball-build.yml#L12-L16) of Linux distribution officially building and testing using source-build.  
+Community building .NET for the [following](https://github.com/dotnet/source-build#net-in-linux-distributions) linux distributions.
+
+**Supported RIDs**
+You can see the list of supported RIDs and the RID graph in the [runtime.json](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.NETCore.Platforms/src/runtime.json). Learn more about RID catalog in [here](https://learn.microsoft.com/en-us/dotnet/core/rid-catalog#linux-rids).
+
+Follow this two stage build process:
+
+**Stage 1:**
+Build bootstrapping .NET SDK and toolset.
+
+- prep.sh (args???)
+ Downloads platform-native version of .NET SDK and toolset. So if you are building on x64 you get x64. Build .NET on the targeted platform to get intended platform-native version of .NET.
+- build.sh (args???)  
+Create source built .NET SDK.
+
+**Stage 2:**
+Use source built .NET SDK and toolset created in stage 1 to build .NET SDK from source. No need to run prep.sh in this stage.
+
+- Build.sh (Args?????)
+
+## Building for New OS (Unsupported)
+
+**RIDs**
+The RID graph or runtime fallback graph is a list of RIDs that are compatible with each other. You can see the list of supported RIDs and the RID graph in the [runtime.json](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.NETCore.Platforms/src/runtime.json). Learn more about RID catalog in [here](https://learn.microsoft.com/en-us/dotnet/core/rid-catalog#linux-rids).
+Determine the compatible supported OS to use as host to build. Choose the host with same processor architecture as that of the new targeted platform.
+
+**Stage 1:**
+Build bootstrapping .NET SDK and toolset.
+
+- prep.sh (args???)
+ Downloads platform-native version of .NET SDK and toolset. So if you are building on x64 you get x64.
+- build.sh (args???)
+Create source built .NET SDK which will be utilized to bootstrap in stage 2.
+
+**Stage 2:**
+Use source built .NET SDK and toolset created in stage 1 to build .NET SDK from source. No need to run prep.sh in this stage.
+
+- Build.sh (Arg?????)
+[UPDATE THE RID GRAPH???]
+
+**If no compatible OS/platform found**
+In stage 1, create portable SDK (this step works for .NET 6, but not for .NET 7). [Define the steps in here???]
+
+## Building for New Architecture (Supported, RID Existing)
+
+**Supported Archs**
+Officially building and testing for the following architectures - x64, Arm64
+Community building for following architectures - x64, Arm64, S390, Arm32, IBM Z
```

Comment:
Do we want to use identifiers (`x64`) or brand/official names (eg, `IBM Z`) here? I think we should pick one style and use that?

S390 (or maybe we can use `s390x` as it's called by Linux or `S390x` as it's called by [.NET API](https://github.com/dotnet/runtime/blob/daa07f8e7ce28e5aba2d32ff586b261839adf91f/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Architecture.cs#L37)) and IBM Z are the same architecture.

We can also add IBM Power (`ppc64le`) here.

Reply:
Updating to use identifiers